blog/2023-11-01-feature-flags-sdks-architectures.md

```diff
@@ -0,0 +1,127 @@
+* Performance and resources usage: each operation like feature flag evaluation causing network traffic.
+    * Time-consuming.
+    * High load on network traffic.
+    * Affecting both application, and Feature Flag service itself, whether it is self-hosted relay proxy or even more when it is a Cloud Saas service.
+
+### API endpoints requests with cache
+
+With this approach, the feature flag service exposes APIs via endpoints like REST HTTP and/or gRpc, and the SDK calling the endpoint requests, and translate the request/responses to the programming language. A cache is used for responses data.  
+Feature flag evaluation is done on proxy / features service only.  
+As this approach involves network traffic for non-cached evaluations, a Relay Proxy is effective for having the network traffic data on internal network for faster and efficient performance and reducing the load on the Features Service / Source.
+
+### API endpoints requests with cache - example flow
+
+<img src={require('@site/static/img/blog/feature-flags-sdks-architectures/API-endpoints-requests-with-cache.png').default} />
+
+### API endpoints requests with cache using Relay Proxy - example flow
+
+<img src={require('@site/static/img/blog/feature-flags-sdks-architectures/API-endpoints-requests-with-cache-using-Relay-Proxy.png').default} />
+
+#### Advantages
+
+* Network traffic can be reduced.
+
+#### Disadvantages
+
+* Functional disadvantage is that intermediate issue like network or feature flags service error for non-cached data is handled by not evaluating to the expected feature flag value.
+* Another functional disadvantage is that intermediate issue like network or feature flags service error during microservice initialization is handled by not evaluating to the expected feature flag value.
+* First requests are not in cache, thus slower and causing network traffic.
+* Not every request can be efficiently cached, as some evaluations can be dynamic according to context with different values.
+* When there is no background task for getting real-time configuration changes event and refreshing the cache, values can take time to be updated.
+
+### Local evaluation
+
+With this approach, the feature flag configuration is saved locally at the SDK.  
+Feature flag evaluation is done locally via the SDK and not affecting network traffic.
+With this approach, configuration can be fetched on initialization, and be refreshed periodically and/or when there is a configuration change, which can be triggered at the SDK via stream event from the feature flag service. 
+Some SDKs sync data via WebSockets or [SSE](https://en.wikipedia.org/wiki/Server-sent_events).
+When this approach is used, evaluation analytics metrics can be aggregated at the SDK side and published separately, as not every flag evaluation triggers a message to the Features Service / Proxy.
```

Comment:
I would recommend leaving out this sentence since you mentioned it was out of scope in the notes.